Mage.Sets/src/mage/cards/t/TheEternalWanderer.java

```diff
@@ -0,0 +1,231 @@
+                CardType.CREATURE.getPredicate()
+        ));
+    }
+
+    public TheEternalWanderer(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.PLANESWALKER}, "{4}{W}{W}");
+        
+        this.addSuperType(SuperType.LEGENDARY);
+        this.setStartingLoyalty(5);
+
+        // No more than one creature can attack The Eternal Wanderer each combat.
+
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new TheEternalWandererAttackRestrictionEffect()));
+
+
+        // +1: Exile up to one target artifact or creature. Return that card to the battlefield under its owner's control at the beginning of that player's next end step. 
+
+        Ability ability = new LoyaltyAbility(new TheEternalWandererExileEffect(), 1);
+        ability.addTarget(new TargetPermanent(0, 1, filter));
+        this.addAbility(ability);
+
+
+
+        // 0: Create a 2/2 white Samurai creature token with double strike.
+
+        this.addAbility(new LoyaltyAbility(new CreateTokenEffect(new DoublestrikeSamuraiToken()), 0));
+
+
+        // -4: For each player, choose a creature that player controls. Each player sacrifices all creatures they control not chosen this way.
+
+        this.addAbility(new LoyaltyAbility(new TheEternalWandererSacrificeEffect(), -4));
+
+    }
+
+    private TheEternalWanderer(final TheEternalWanderer card) {
+        super(card);
+    }
+
+    @Override
+    public TheEternalWanderer copy() {
+        return new TheEternalWanderer(this);
+    }
+}
+
+class TheEternalWandererExileEffect extends OneShotEffect {
+
+    public TheEternalWandererExileEffect() {
+        super(Outcome.Detriment);
+        staticText = "exile up to one target artifact or creature. If you do, return that card to the battlefield under its owner's control at the beginning of the next end step";
+    }
+
+    public TheEternalWandererExileEffect(final TheEternalWandererExileEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        MageObject sourceObject = source.getSourceObject(game);
+        if (controller != null && sourceObject != null) {
+            Permanent permanent = game.getPermanent(getTargetPointer().getFirst(game, source));
+            System.out.println(permanent);
+            if (permanent != null) {
+                UUID exileId = UUID.randomUUID();
+                if (controller.moveCardsToExile(permanent, source, game, true, exileId, sourceObject.getIdName())) {
+                    //create delayed triggered ability
+                    Effect effect = new ReturnToBattlefieldUnderOwnerControlTargetEffect(false, false);
+                    effect.setTargetPointer(new FixedTarget(permanent.getId(), game));
+                    game.addDelayedTriggeredAbility(new AtTheBeginOfNextEndStepDelayedTriggeredAbility(effect), source);
```

Comment:
This delayed triggered ability triggers at the beginning of the *next* end step, not the beginning of the *card's owner's* next end step.

Reply:
fixed the exile effect to use new AtTheBeginOfPlayersNextEndStepDelayedTriggeredAbility which takes in a target player (in this case, the owner of the artifact/creature that is being exiled) then returns it during that players next end step